src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -187,10 +187,18 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                         OutputMessage.Severity.Warning));
                 }
 
-                ITransport localTransport = null;
+                ITransport localTransport;
+
+                //Attempt to support RunInTerminal first
+                if (HostRunInTerminal.IsRunInTerminalAvailable()
+                    && String.IsNullOrWhiteSpace(localLaunchOptions.MIDebuggerServerAddress)
+                    && IsCoreDump == false)
+                {
+                    localTransport = new RunInTerminalTransport();
+                }
                 // For local Linux and OS X launch, use the local Unix transport which creates a new terminal and
                 // uses fifos for debugger (e.g., gdb) communication.
-                if (this.MICommandFactory.UseExternalConsoleForLocalLaunch(localLaunchOptions) &&
+                else if (this.MICommandFactory.UseExternalConsoleForLocalLaunch(localLaunchOptions) &&
```

Comment:
I feel like we should remove or refactor `UseExternalConsoleForLocalLaunch` since its logic is too redundant with deciding if we should use RunInTerminal transport. So we would be better off either change that method to instead decide which of the three local transports to use, or just delete it all together and move the checks that make sense here.